src/currencyConversion/BitmamaAPI.ts

```diff
@@ -0,0 +1,60 @@
+import { RESTDataSource } from 'apollo-datasource-rest'
+import BigNumber from 'bignumber.js'
+import { logger } from '../logger'
+import { BITMAMA_API } from '../config'
+
+interface BitmamaAPIResult {
+  buy: number
+  sell: number
+}
+
+// ttl in seconds!
+const MIN_TTL = 12 * 3600 // 12 hours
+export default class BitmamaAPI extends RESTDataSource {
+  private rateCache: Map<string, BigNumber>
+  constructor() {
+    super()
+    this.baseURL = BITMAMA_API
+    this.rateCache = new Map<string, BigNumber>()
```

Comment:
```suggestion
    this.rateCache = new Map<string, BigNumber>() // should be ok to not have an eviction policy because we only use this for USD/Naira, and even if we used all 180 currencies in the world, the number of entries has a manageable upper bound
```
I had an argument with myself over whether the potential for a memory leak here was acceptable. I think it's ok, but I suggest we leave this comment here to explain the rationale